python3/httplib2/__init__.py

```diff
@@ -152,6 +152,10 @@ def _build_ssl_context(
         raise RuntimeError("httplib2 requires Python 3.2+ for ssl.SSLContext")
 
     context = ssl.SSLContext(DEFAULT_TLS_VERSION)
+    # It is required to disable check_hostname validation before changing verify_mode to CERT_NONE
+    # for PROTOCOL_TLS_CLIENT
+    if DEFAULT_TLS_VERSION.name == 'PROTOCOL_TLS_CLIENT':
+        context.check_hostname = not bool(disable_ssl_certificate_validation)
```

Comment:
```suggestion
    context.check_hostname = not disable_ssl_certificate_validation
```

check_hostname is 3.4+, condition seems redundant

Reply:
It turned out that the condition is required. 
It seems that there were some breaking changes with check_hostname - so for python3.5 it should be disabled after disabling verify_mode and for others conversely.

I run CI on my fork and it seems that now tests are [passing](https://github.com/ziollek/httplib2/actions/runs/3489726352).

Reply:
you are right - check_hostname should be set after verify_mode

without condition

Reply:
sorry, it was a bit more complicated, interesting reading here https://bugs.python.org/issue31431 (same) https://github.com/python/cpython/issues/75612

new version works on 3.5, 3.8 and 3.10